LuatixHQ/opencti
# [frontend/backend] In CSV Feed Ingester, take into account Default Marking definition options from CSV Mapper

### Proposed changes

* handle CSV mapper configuration on CSV feed ingester creation
* handle CSV mapper configuration on CSV feed ingester edition
* update Launch import dialogs

### Related issues

* https://github.com/OpenCTI-Platform/opencti/issues/6149

### Checklist







### Further comments



To test this feature, create CSV mappers with both options: 
- Let the user choose
- Use default marking definitions of the user

![image](https://github.com/OpenCTI-Platform/opencti/assets/13460700/a3336a38-ca41-4ffa-b9e0-323411c5db05)

![image](https://github.com/OpenCTI-Platform/opencti/assets/13460700/e088938b-4478-4137-a6f8-5388fe07f549)

![image](https://github.com/OpenCTI-Platform/opencti/assets/13460700/355bf102-f752-4708-a15d-33eb53e813ad)

![image](https://github.com/OpenCTI-Platform/opencti/assets/13460700/aebc376e-ed61-4272-b4d2-597dec0917fb)

To set default markings:
- you have to go to Customization > Entity types
![image](https://github.com/OpenCTI-Platform/opencti/assets/13460700/33785f41-b675-41cf-861a-7eca14b33c31)

- select an entity type
- select the Marking attribute
![image](https://github.com/OpenCTI-Platform/opencti/assets/13460700/5b95c6a7-57f7-4829-b2d2-954d27c41b02)

- activate default values
![image](https://github.com/OpenCTI-Platform/opencti/assets/13460700/3099a992-f714-439e-835a-63423a474740)

- navigate to Security > Groups, select the current group (=> default)
![image](https://github.com/OpenCTI-Platform/opencti/assets/13460700/37cd2165-753c-4117-9a64-dcdc6c4a90fb)

- select allowed markings, so they can be added to the list of options for default markings
- select default markings
![image](https://github.com/OpenCTI-Platform/opencti/assets/13460700/d64672f7-1c37-477d-ad85-a2723b3279e3)

Single column CSV file to test your CVS mapper: 
[6149-ipv4-addr.csv](https://github.com/user-attachments/files/15805659/6149-ipv4-addr.csv)

Single column CSV feed to ingest IPv4 addresses: https://lists.blocklist.de/lists/all.txt

To check for the successful ingestion: 
- make sure your worker is running
- go to Observations > Observables
- see the IPv4 addresses with corresponding markings

To check different use cases :
- purge messages in push_sync queue in RabbitMQ
![image](https://github.com/OpenCTI-Platform/opencti/assets/13460700/dee838c3-9fad-4d26-8dc2-a5f14b6f02be)
- delete ingested entities in elastic console
```
POST /opencti_stix_cyber_observables-000001/_delete_by_query
{
  "query": {
    "bool": {
      "must": [
          { "term":{"entity_type.keyword": "IPv4-Addr" }}
        ]
    } // match_all
  }
}
```
![image](https://github.com/OpenCTI-Platform/opencti/assets/13460700/f77cec99-b1da-4cab-b937-e3ca5e7a5efc)
